### ISSUE TITLE
Supporting lists of functions in map_blocks/map_overlap

### ISSUE THREAD

**Issue by jakirkham** [dask/dask#3547]
Would be nice for function composition to be able to provide a list of functions that will be applied in order in a single `map_blocks` or `map_overlap` operation. This is useful for reducing overhead involved with `map_overlap` in particular, but would be nice to use with `map_blocks` as well.

cc @jni @mrocklin

**Comment by TomAugspurger** [comment]
@jakirkham do you think this is still relevant now that high-level graph fusion is in?

**Comment by mrocklin** [comment]
Yes, this is likely still relevant, particularly for map_overlap.

**Comment by jakirkham** [comment]
Yeah, this is helpful as people can cutdown on repeated communication between neighboring chunks in a computation by simply accepting a larger overlap at the outset.

I don't think high-level graph fusion addresses this currently and it may be tricky for it to do, but would be happy if someone proved me wrong 😄
